### PR TITLE
feat(lib): Add ability to set where script is appended

### PR DESCRIPTION
**Pull request by brdlyptrs**
Closes https://github.com/hCaptcha/react-hcaptcha/issues/183

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hcaptcha/react-hcaptcha",
-  "version": "1.6.1",
+  "version": "1.6.2",
```
Comment (zoryana94) [inline package.json:3]: it's not actually a bugfix, it's an update
so it should be `1.7.0`

File: tests/hcaptcha.spec.js [hunk 2]
```diff
@@ -544,5 +531,112 @@ describe("hCaptcha", () => {
...
+                expect(script).toBeTruthy();
+            });
+
+            it("should append script into specified DOM element", () => {
+                const element = document.createElement('div');
+                element.id = "script-location";
+
+                document.body.appendChild(element);
+
+                const nstance = ReactTestUtils.renderIntoDocument(<HCaptcha
```
Comment (zoryana94) [inline tests/hcaptcha.spec.js:553]: typo `nstance` => `instance`

File: tests/hcaptcha.spec.js [hunk 3]
```diff
@@ -544,5 +531,112 @@ describe("hCaptcha", () => {
             const script = document.querySelector("head > script");
             expect(script.async).toBeTruthy();
         });
+
+        describe("scriptLocation", () => {
+
+
+             it("validate src without", () => {
```
Comment (zoryana94) [inline tests/hcaptcha.spec.js:538]: how about "should be document.head by default"?

File: tests/hcaptcha.spec.js [hunk 4]
```diff
@@ -544,5 +531,112 @@ describe("hCaptcha", () => {
...
+
+            describe('iframe', () => {
+                const iframe = document.createElement('iframe');
+                 document.body.appendChild(iframe);
+
+                const iframeDoc = iframe.contentWindow.document;
+
+                afterAll(() => {
+                    // Clean up
+                    document.body.removeChild(iframe);
```
Comment (zoryana94) [inline tests/hcaptcha.spec.js:590]: you may want to use `afterEach` here, so that it cleans up after each test

**Comment by zoryana94** [comment]
The types have been missed to be updated

File: README.md [hunk 5]
```diff
@@ -139,7 +139,8 @@ return <HCaptcha ref={captchaRef} onLoad={onLoad} sitekey={sitekey} {...props} /
 |`reportapi`|String|No|`-`|See enterprise docs.|
 |`sentry`|String|No|`-`|See enterprise docs.|
 |`custom`|Boolean|No|`-`|See enterprise docs.|
-|`loadAsync`|Boolean|No|`true`|Set if the script should be loaded asynchronously|
+|`loadAsync`|Boolean|No|`true`|Set if the script should be loaded asynchronously.|
+|`scriptLocation`|Element|No|`true`|Location of where to append the script tag.|
```
Comment (zoryana94) [inline README.md:143]: the default scriptLocation is document.head

File: src/index.js [hunk 6]
```diff
@@ -14,7 +14,12 @@ const mountPromise = new Promise((resolve, reject) => {
 
 // Generate hCaptcha API script
 const mountCaptchaScript = (params={}) => {
-  if (document.getElementById(SCRIPT_ID)) {
+  const parent = params.scriptLocation || document.head;
```
Comment (zoryana94) [inline src/index.js:17]: what about `scriptContainer`?

**Review by zoryana94 — DISMISSED** [review]

File: tests/hcaptcha.spec.js (hunk 3 above)
Comment (brdlyptrs) [reply tests/hcaptcha.spec.js]: Moved this test, to this area. Just updated it though here [e71a40d](https://github.com/hCaptcha/react-hcaptcha/pull/182/commits/e71a40df0f0d823d608e055f4ff7d16fc6f308e8)

File: tests/hcaptcha.spec.js (hunk 2 above)
Comment (brdlyptrs) [reply tests/hcaptcha.spec.js]: Removed variable, as it is not needed

**Review by zoryana94 — DISMISSED** [review]

**Review by zoryana94 — APPROVED** [review]
👍

**Pull request merged**